libraries/AP_SensorHead/Protocol.h

```diff
@@ -0,0 +1,182 @@
+#pragma once
+
+#include <AP_Math/AP_Math.h>
+
+namespace SensorHead {
+
+    enum class msgid_t : uint8_t;
+
+    /* Packet structure */
+    class Packet {
+    public:
+        static const uint8_t MAGIC = 0xFC;
+        static const uint8_t VERSION = 0;
+
+        typedef struct PACKED {
+            uint8_t magic;
+            uint8_t ver;
+            uint8_t len;
+            msgid_t id;
+        } shead_hdr_t;
+
+        typedef uint16_t shead_crc_t;
+
+        typedef struct PACKED {
+            shead_hdr_t *hdr;
```

Comment:
i'm surprised at the use of pointers in this packet format - does it end up doing 3 allocations, or are the pointers into fixed arrays?